src/sql/azdata.proposed.d.ts

```diff
@@ -360,6 +360,37 @@ declare module 'azdata' {
 		}
 
 		export function createModelViewDashboard(title: string, options?: ModelViewDashboardOptions): ModelViewDashboard;
+
+		export interface Dialog {
+			/**
+			 * Width of the dialog
+			 */
+			width?: DialogWidth;
+		}
+
+		export interface Wizard {
+			/**
+			 * Width of the wizard
+			 */
+			width?: DialogWidth;
+		}
+
+		export type DialogWidth = 'narrow' | 'medium' | 'wide' | number;
+
+		/**
+		 * Create a dialog with the given title
+		 * @param title The title of the dialog, displayed at the top
+		 * @param dialogName the name of the dialog
+		 * @param width width of the dialog, default is 'wide'
+		 */
+		export function createModelViewDialog(title: string, dialogName?: string, width?: DialogWidth): Dialog;
```

Comment:
We should figure out how we want to handle deprecating stuff - but I feel like a @deprecated annotation on the azdata.d.ts one would be a good idea.